### PR TITLE
move buildCyclonedxLib to Build the CycloneDX Java library and app used for SBoM generation from build.sh to sbom.sh

### PR DESCRIPTION
**Pull request by julian55455**
move buildCyclonedxLib to Build the CycloneDX Java library and app used for SBoM generation from build.sh to sbom.sh

Ref: https://github.com/adoptium/temurin-build/issues/3158

### CODE REVIEW

**Comment by karianna** [comment]
@dependabot recreate

**Comment by karianna** [comment]
@julian55455 you'll need to sign the Eclipse CLA?

**Comment by karianna** [comment]
@julian55455 - Also linter and tests fail

**Review by zdtsw — APPROVED** [review]

**Comment by zdtsw** [comment]
you might need to do a rebase first

**Review by andrew-m-leonard — APPROVED** [review]

**Comment by zdtsw** [comment]
will merge it once code frozen is uplifted

**Comment by zdtsw** [comment]
@julian55455 need to resolve conflicts first ^

**Comment by julian55455** [comment]
@zdtsw are these build failures related to the changes made in this PR?

**Comment by zdtsw** [comment]
> @zdtsw are these build failures related to the changes made in this PR?

dont think so, but since both mac failed, i will run one of them and see if it was glitch or not

**Comment by andrew-m-leonard** [comment]
I don't think the CycloneDX code is getting built correctly, but only Mac shows an error for some reason

File: sbin/build.sh [hunk 1]
```diff
@@ -968,17 +946,17 @@ cleanAndMoveArchiveFiles() {
 
   echo "Currently at '${PWD}'"
 
-  local jdkPath=$(ls -d ${BUILD_CONFIG[JDK_PATH]})
+  local jdkPath=$(ls -d "${BUILD_CONFIG[JDK_PATH]}")
```
Comment (andrew-m-leonard) [inline sbin/build.sh:949]: This line needs the quotes removing as it relies on glob expansion.
Please can you undo the other " changes above as well

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

**Comment by zdtsw** [comment]
lets try get this pass test first and merge it, leave the "linter" to issue https://github.com/adoptium/ci-jenkins-pipelines/issues/428 to fix

**Comment by zdtsw** [comment]
> Ah of course the github action does not use --create-sbom... must be a glitch ?

think the reason is, we wanna only enable sbom related part from ci-jenkins-pipeline, and decouple from temurin-build.
so the others fork the repo wont need to do anything with sbom.

if we want sbom be in the GH check here, we later will also need to add more logic for the signing keys, can be a bit complicated.

**Comment by andrew-m-leonard** [comment]
@julian55455 For the linter error:
```
Error: File:[/github/workspace/sbin/build.sh] is not executable
Error: File:[/github/workspace/sbin/common/sbom.sh] is not executable
```
Try doing a new commit with:
chmod +x sbin/build.sh
chmod +x sbin/common/sbom.sh

You may need to do an arbitrary change in each file to make a commit.

File: sbin/common/sbom.sh [hunk 2]
```diff
@@ -1,4 +1,32 @@
 #!/bin/bash
+# Build the CycloneDX Java library and app used for SBoM generation
+buildCyclonedxLib() {
+  local javaHome="${1}"
+
+  # Check if JDK-17 is available, download it if it isn't
+  if [ ! -d "${javaHome}" ]; then
+    # Download JDK-17
+    apiUrlTemplate="https://api.adoptium.net/v3/binary/latest/\${JDK_BOOT_VERSION}/\${releaseType}/linux/\${downloadArch}/jdk/hotspot/normal/\${vendor}"
```
Comment (andrew-m-leonard) [inline sbin/common/sbom.sh:9]: You don't need these 2 lines:
apiUrlTemplate=...
apiURL=$(eval echo ${apiUrlTemplate})

File: sbin/common/sbom.sh [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+    apiURL=$(eval echo ${apiUrlTemplate})
+    JDK_BOOT_VERSION="17"
```
Comment (andrew-m-leonard) [inline sbin/common/sbom.sh:11]: Make all these "local" variables

File: sbin/common/sbom.sh [hunk 4]
```diff
@@ -1,4 +1,32 @@
 #!/bin/bash
+# Build the CycloneDX Java library and app used for SBoM generation
+buildCyclonedxLib() {
+  local javaHome="${1}"
+
+  # Check if JDK-17 is available, download it if it isn't
```
Comment (andrew-m-leonard) [inline sbin/common/sbom.sh:6]: Move this boot jdk logic to https://github.com/adoptium/temurin-build/blob/18f343c90c72084cad7f6d29f14721e1309ed1ac/sbin/build.sh#L688

File: sbin/common/sbom.sh [hunk 5]
```diff
@@ -1,4 +1,32 @@
...
+  if [ ! -d "${javaHome}" ]; then
+    # Download JDK-17
+    apiUrlTemplate="https://api.adoptium.net/v3/binary/latest/\${JDK_BOOT_VERSION}/\${releaseType}/linux/\${downloadArch}/jdk/hotspot/normal/\${vendor}"
+    apiURL=$(eval echo ${apiUrlTemplate})
+    JDK_BOOT_VERSION="17"
+    releaseType="ga"
+    downloadArch="x64"
+    vendor="adoptium"
+    apiURL="https://api.adoptium.net/v3/binary/latest/${JDK_BOOT_VERSION}/${releaseType}/linux/${downloadArch}/jdk/hotspot/normal/${vendor}"
+    echo "Downloading GA release of boot JDK version ${JDK_BOOT_VERSION} from ${apiURL}"
```
Comment (andrew-m-leonard) [inline sbin/common/sbom.sh:16]: You are missing the actual download..
```
curl -L "${apiURL}" | tar xpzf - --strip-components=1 -C "$jdkDir"
```
Set jdkDir variable to be "${CYCLONEDB_DIR}/jdk"
Also add that folder to https://github.com/adoptium/temurin-build/blob/master/.gitignore

Comment (andrew-m-leonard) [reply sbin/common/sbom.sh:16]: Then set javaHome in setupAntEnv()

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

File: sbin/common/sbom.sh (hunk 4 above)
Comment (zdtsw) [reply sbin/common/sbom.sh:6]: agree, i was about to say the logic should be in `setupAntEnv() `

**Comment by julian55455** [comment]
fixed in this pr https://github.com/adoptium/temurin-build/pull/3243

**Pull request closed without merging**